py_ballisticcalc/trajectory_calc.py

```diff
@@ -179,95 +181,111 @@ def _zero_angle(self, ammo: Ammo, weapon: Weapon, atmo: Atmo):
-        length = ammo.length >> Distance.Inch
+        look_angle = shot_info.look_angle >> Angular.Radian
+        twist = shot_info.weapon.twist >> Distance.Inch
+        length = ammo.dm.length >> Distance.Inch
         diameter = ammo.dm.diameter >> Distance.Inch
         weight = ammo.dm.weight >> Weight.Grain
 
-        # step = shot_info.step >> Distance.Foot
         step = dist_step >> Distance.Foot
         calc_step = self.get_calc_step(step)
 
-        maximum_range = (shot_info.max_range >> Distance.Foot) + 1
+        maximum_range = (max_range >> Distance.Foot) + 1
 
         ranges_length = int(maximum_range / step) + 1
         len_winds = len(winds)
         current_wind = 0
         current_item = 0
-
+        next_range_distance = .0
+        previous_mach = .0
+        ranges = []
+        drag = 0
         stability_coefficient = 1.0
+        twist_coefficient = 0
         next_wind_range = 1e7
-
-        barrel_elevation = (shot_info.zero_angle >> Angular.Radian) + (
-                shot_info.relative_angle >> Angular.Radian)
         alt0 = atmo.altitude >> Distance.Foot
-        sight_height = weapon.sight_height >> Distance.Foot
 
-        next_range_distance = .0
-        barrel_azimuth = .0
+        barrel_elevation = shot_info.barrel_elevation >> Angular.Radian
+        barrel_azimuth = shot_info.barrel_azimuth >> Angular.Radian
+        sight_height = shot_info.weapon.sight_height >> Distance.Foot
+        cant_cosine = math.cos(shot_info.cant_angle >> Angular.Radian)
+        cant_sine = math.sin(shot_info.cant_angle >> Angular.Radian)
+        range_vector = Vector(.0, -cant_cosine*sight_height, -cant_sine*sight_height)
 
         gravity_vector = Vector(.0, cGravityConstant, .0)
-        range_vector = Vector(.0, -sight_height, .0)
-
-        ranges = []
-
-        previous_y = 0
-        previous_mach = 0
 
         if len_winds < 1:
             wind_vector = Vector(.0, .0, .0)
         else:
             if len_winds > 1:
                 next_wind_range = winds[0].until_distance() >> Distance.Foot
-            wind_vector = wind_to_vector(shot_info, winds[0])
+            wind_vector = wind_to_vector(winds[0])
 
         if Settings.USE_POWDER_SENSITIVITY:
             velocity = ammo.get_velocity_for_temp(atmo.temperature) >> Velocity.FPS
         else:
             velocity = ammo.mv >> Velocity.FPS
 
-        # x - distance towards target, y - drop and z - windage
+        # x: downrange distance (towards target), y: drop, z: windage
         velocity_vector = Vector(math.cos(barrel_elevation) * math.cos(barrel_azimuth),
                                  math.sin(barrel_elevation),
                                  math.cos(barrel_elevation) * math.sin(barrel_azimuth)) * velocity
 
         if twist != 0 and length and diameter:
-            stability_coefficient = calculate_stability_coefficient(ammo, weapon, atmo)
-            twist_coefficient = -1 if twist > 0 else 1
```

Comment:
Why we're comming back to this again? As I know we're already done with this, and the direction that was here is valid

Reply:
Must have gotten reverted somewhere along the way.  I have several unit tests for spin drift in test_computer.py, so we can be sure of this going forward.